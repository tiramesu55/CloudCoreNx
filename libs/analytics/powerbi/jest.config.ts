/* eslint-disable */
export default {
  displayName: 'analytics-powerbi',
  preset: '../../../jest.preset.js',
  transform: {
    '^.+\\.[tj]sx?$': 'babel-jest',
  },
  moduleFileExtensions: ['ts', 'tsx', 'js', 'jsx'],
  coverageDirectory: '../../../coverage/libs/analytics/powerbi',
};
