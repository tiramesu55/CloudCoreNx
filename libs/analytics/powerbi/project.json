{
  "$schema": "../../../node_modules/nx/schemas/project-schema.json",
  "sourceRoot": "libs/analytics/powerbi/src",
  "projectType": "library",
  "tags": [],
  "targets": {
    "lint": {
      "executor": "@nrwl/linter:eslint",
      "outputs": ["{options.outputFile}"],
      "options": {
        "lintFilePatterns": ["libs/analytics/powerbi/**/*.{ts,tsx,js,jsx}"]
      }
    },
    "test": {
      "executor": "@nrwl/jest:jest",
      "outputs": ["coverage/libs/analytics/powerbi"],
      "options": {
        "jestConfig": "libs/analytics/powerbi/jest.config.ts",
        "passWithNoTests": true
      }
    }
  }
}
