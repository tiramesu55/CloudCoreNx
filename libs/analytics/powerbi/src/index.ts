export * from './lib/analytics-powerbi';
export * from './lib/components/index';
