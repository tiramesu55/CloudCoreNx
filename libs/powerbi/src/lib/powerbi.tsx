import { Route, Link } from 'react-router-dom';

import styles from './powerbi.module.css';

/* eslint-disable-next-line */
export interface PowerbiProps {}

export function Powerbi(props: PowerbiProps) {
  return (
    <div className={styles['container']}>
      <h1>Welcome to Powerbi!</h1>

      <ul>
        <li>
          <Link to="/">powerbi root</Link>
        </li>
      </ul>
      {/* <Route path="/" element={<div>This is the powerbi root route.</div>} /> */}
    </div>
  );
}

export default Powerbi;
