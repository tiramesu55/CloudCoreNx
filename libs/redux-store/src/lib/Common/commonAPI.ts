import axios from 'axios';

export const getUserConfigApi = (url: string, token: string) =>
  axios.get(`${url}/UserConfiguration`, {
    headers: {
      Authorization: `Bearer ${token}`,
    },
  });

export const upsertUserConfigApi = (url: string, token: string, data: any) =>
  axios.put(`${url}/UserConfiguration`, data, {
    headers: {
      Authorization: `Bearer ${token}`,
    },
  });
