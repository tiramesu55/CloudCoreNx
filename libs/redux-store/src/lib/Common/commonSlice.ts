import { createSlice, PayloadAction } from '@reduxjs/toolkit';

export interface CommonState {
  openAlert: boolean;
  type: 'success' | 'info' | 'error' | 'warning';
  content: string;
}

interface ActionPayload {
  type: 'success' | 'info' | 'error' | 'warning';
  content: string;
}

const initialState: CommonState = {
  openAlert: false,
  type: 'error',
  content: '',
};

export const commonSlice = createSlice({
  name: 'common',
  initialState,
  reducers: {
    openAlertAction: (state, action: PayloadAction<ActionPayload>) => {
      state.openAlert = true;
      //  state.applications
      state.content = action.payload.content;
      state.type = action.payload.type;
    },
    closeAlertAction: (state) => {
      state.openAlert = false;
      //  state.applications
      state.content = '';
      // state.type = 'error';
    },
  },
});
export const { openAlertAction, closeAlertAction } = commonSlice.actions;
export const commonReducer = commonSlice.reducer;
