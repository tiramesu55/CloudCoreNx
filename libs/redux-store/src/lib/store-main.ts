import { configureStore, ThunkAction, Action } from "@reduxjs/toolkit";
import { reportReducer } from "./PowerBI/reducers/reportReducer"
import {userReducer} from './Platfotm-UI/users/userSlice';
import {applicationsReducer} from './Platfotm-UI/applications/applicationsSlice';
import {sitesReducer} from './Platfotm-UI/sites/siteSlice';
import {dashboardReduser} from './Platfotm-UI/dashboard/dashboardSlice';
import { organizationsReducer } from './Platfotm-UI/organizations/organizationsSlice';
import { TypedUseSelectorHook,  useSelector, useDispatch } from 'react-redux';
export const store = configureStore({
   reducer:{
      report: reportReducer,
      user: userReducer,
      applications: applicationsReducer,
      sites: sitesReducer,
      dashboard: dashboardReduser,
      organizations: organizationsReducer
   },
   devTools: process.env['NODE_ENV'] !== 'production',
   // Optional Redux store enhancers
   enhancers: [],
})
//type for combine state
export type RootState = ReturnType<typeof store.getState>
export type AppDispatch = typeof store.dispatch;
export type AppThunk<ReturnType = void> = ThunkAction<
  ReturnType,
  RootState,
  unknown,
  Action<string>
>;
export const useAppSelector: TypedUseSelectorHook<RootState> = useSelector;
export const useAppDispatch = () => useDispatch<AppDispatch>();