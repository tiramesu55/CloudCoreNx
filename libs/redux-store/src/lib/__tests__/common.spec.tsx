/* eslint-disable @nrwl/nx/enforce-module-boundaries */
/* eslint-disable @typescript-eslint/no-empty-function */
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom'
import { act } from 'react-dom/test-utils';   
import { configureStore, Action } from '@reduxjs/toolkit';
import { ThunkAction } from 'redux-thunk';
import { CommonState, commonSlice, getUserConfig, upsertUserConfig, UserConfig } from '../Common/commonSlice';
import axios, { AxiosResponse } from 'axios';
import { StrictMode } from 'react';

import { Provider } from 'react-redux';
import { store } from '../store-main';
import { BrowserRouter } from 'react-router-dom';
import { AnalyticsPowerbi } from '@cloudcore/analytics/powerbi';
import { ConfigCtx, IConfig, OktaCode, ConfigContext } from '@cloudcore/okta-and-config';
import { useContext } from 'react';
import { createTheme, ThemeProvider } from '@mui/material';
import { theme } from '@cloudcore/ui-shared';

import { config, userData, MaintenanceDetails, PlatformApplication, GetSuitesByPermission } from './mocks/index'; 

jest.mock('axios');
// const errorMessage = 'Error retrieving user configuration';
//     axios.get.mockRejectedValue(new Error(errorMessage));


// interface UserConfig {
//     discriminator: string;
//     email: string;
//     id: string;
//     lastOpenedReportId: string;
//     userSetDefaultApp: string;
//     defaultApp: null | string;
//     orgCode: string;
//     inactiveDate: Date | null;
//     startDate: Date | null;
//     endDate: Date | null;
//     createdBy: string;
//     modifiedBy: string;
//     createdDate: Date | null;
//     modifiedDate: Date | null;
//   }
// const store = configureStore({
//     reducer: {
//         common: commonSlice.reducer,
//     },
// });
beforeEach(() => {
    (axios.get as jest.Mock).mockReset();
  });

describe('commonSlice', () => {
    it('should render AnalyticsPowerbi component', async () => {

        (axios.get as jest.Mock).mockImplementation(async (url) => {
            console.log('url', url)
            if (url.indexOf('config') !== -1) {
              return Promise.resolve(config);
            } else if (url.indexOf('GetSuitesByPermission') !== -1) {
                 return Promise.resolve( {data :GetSuitesByPermission});
            } else if (url.indexOf('Maintenance') !== -1) {
                return Promise.resolve({data: MaintenanceDetails});
            } else if (url.indexOf('PlatformApplication') !== -1) {
                return Promise.resolve({data: PlatformApplication});
            } else if (url.indexOf('PlatformUser') !== -1) {
                return Promise.resolve({data: userData});
            } else {
              throw new Error(`Unexpected URL: ${url}`);
            }
          });
      
        //   const result1 = await myFunctionThatCallsAxios(url1);
        //   expect(result1).toEqual(response1.data);
        //   expect(axios.get).toHaveBeenCalledWith(url1);
      
        //   const result2 = await myFunctionThatCallsAxios(url2);
        //   expect(result2).toEqual(response2.data);
        //   expect(axios.get).toHaveBeenCalledWith(url2);

         let container;
        await act(async () => {
            container = render(<ConfigContext>
                <Provider store={store}>
                    <ThemeProvider theme={theme}>
                        <BrowserRouter>
                        <AnalyticsPowerbi />
                        </BrowserRouter>
                    </ThemeProvider>
                </Provider>
            </ConfigContext>)
        });
        
        expect(container).toMatchSnapshot();

        });
    // });

  it('should handle getUserConfig', async () => {
    const result = await store.dispatch(getUserConfig({ url: 'url', token: 'token' }));
    expect(commonSlice.reducer(store.getState().common, result)).toEqual({
      openAlert: false,
      type: 'error',
      status: 'idle',
      content: '',
      error: '',
      userConfig: {
        discriminator: '',
        email: '',
        id: '',
        lastOpenedReportId: '',
        userSetDefaultApp: '',
        defaultApp: null,
        orgCode: '',
        inactiveDate: null,
        startDate: null,
        endDate: null,
        createdBy: '',
        modifiedBy: '',
        createdDate: null,
        modifiedDate: null,
      },
    });
  });

  it('should handle upsertUserConfig', async () => {
    const data = {
      discriminator: '',
      email: 'test@test.com',
      id: '',
      lastOpenedReportId: '',
      userSetDefaultApp: '',
      defaultApp: null,
      orgCode: '',
      inactiveDate: null,
      startDate: null,
      endDate: null,
      createdBy: '',
      modifiedBy: '',
      createdDate: null,
      modifiedDate: null,
    };
    const url = 'http://example.com';
    const token = 'abc123';
    // const data = { id: '123', name: 'Test User' };
    (axios.get as jest.Mock).mockResolvedValue({ data } as AxiosResponse<UserConfig>);

    const result = await store.dispatch(upsertUserConfig({ url, token, data }));
    expect(commonSlice.reducer(store.getState().common, result)).toEqual({
      openAlert: false,
      type: 'error',
      status: 'idle',
      content: '',
      error: '',
      userConfig: {
        discriminator: '',
        email: '',
        id: '',
        lastOpenedReportId: '',
        userSetDefaultApp: '',
        defaultApp: null,
        orgCode: '',
        inactiveDate: null,
        startDate: null,
        endDate: null,
        createdBy: '',
        modifiedBy: '',
        createdDate: null,
        modifiedDate: null,
      },
    });
  });
});

