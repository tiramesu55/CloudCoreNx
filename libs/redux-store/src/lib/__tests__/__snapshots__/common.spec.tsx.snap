// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`commonSlice should render AnalyticsPowerbi component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
       
      <div
        style="height: 100vh; background: rgb(246, 245, 247);"
      >
        <div />
        <div
          class="MuiBox-root css-0"
        >
          <header
            class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionRelative css-c0mv2e-MuiPaper-root-MuiAppBar-root"
            style="border: 1px; border-color: #0000000d; border-bottom: 1px solid #0000000D; margin: 0px; box-shadow: rgb(0 0 0 / 20%) 0px 2px 4px -1px, rgb(0 0 0 / 14%) 0px 4px 5px 0px, rgb(0 0 0 / 12%) 0px 1px 10px 0px; background: rgb(255, 255, 255); z-index: 1201; font-family: \\"Roboto\\", sans-serif; width: 100%;"
          >
            <div
              class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-bw3mp6-MuiToolbar-root"
              style="padding-left: 20px; padding-right: 20px;"
            >
              <div
                class="MuiBox-root css-17zsna1"
              >
                <div
                  class="MuiTypography-root MuiTypography-h6 MuiTypography-noWrap css-1m9344j-MuiTypography-root"
                >
                  <a
                    aria-current="page"
                    class="active"
                    href="/"
                  >
                    <img
                      alt="NexiaLogo"
                      src="test"
                      style="width: 100%; max-width: 163px; max-height: 41px;"
                    />
                  </a>
                </div>
                <hr
                  class="MuiDivider-root MuiDivider-middle MuiDivider-vertical MuiDivider-flexItem css-my8t4q-MuiDivider-root"
                />
                <div
                  class="MuiBox-root css-bqedkg"
                  style="text-decoration: none; color: rgb(101, 19, 240); font-size: 18px; margin-left: 24px;"
                >
                  Enterprise Analytics
                </div>
                <div
                  class="MuiBox-root css-k008qs"
                >
                  <div
                    class="makeStyles-menuItems-2 MuiBox-root css-1opalb8"
                  >
                    <button
                      aria-expanded="false"
                      aria-haspopup="menu"
                      class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-ipszf5-MuiButtonBase-root-MuiButton-root"
                      tabindex="0"
                      type="button"
                    >
                      Shipping Insights
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                        data-testid="KeyboardArrowDownIcon"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M7.41 8.59 12 13.17l4.59-4.58L18 10l-6 6-6-6 1.41-1.41z"
                        />
                      </svg>
                      <span
                        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                      />
                    </button>
                    <div
                      class="dropdown browserSpecific MuiBox-root css-1hyfx7x"
                    >
                      <button
                        class="subMenuList MuiBox-root css-0"
                        id="SLA Dashboard"
                        type="button"
                      >
                        <div
                          class="MuiBox-root css-17zsna1"
                        >
                          <p
                            class="MuiTypography-root MuiTypography-body1 css-e000bs-MuiTypography-root"
                          >
                            SLA Dashboard
                          </p>
                        </div>
                      </button>
                      <button
                        class="subMenuList MuiBox-root css-0"
                        id="SLA Report"
                        type="button"
                      >
                        <div
                          class="MuiBox-root css-17zsna1"
                        >
                          <p
                            class="MuiTypography-root MuiTypography-body1 css-e000bs-MuiTypography-root"
                          >
                            SLA Report
                          </p>
                        </div>
                      </button>
                      <button
                        class="subMenuList MuiBox-root css-0"
                        id="Service Level Dashboard"
                        type="button"
                      >
                        <div
                          class="MuiBox-root css-17zsna1"
                        >
                          <p
                            class="MuiTypography-root MuiTypography-body1 css-e000bs-MuiTypography-root"
                          >
                            Service Level Dashboard
                          </p>
                        </div>
                      </button>
                      <button
                        class="subMenuList MuiBox-root css-0"
                        id="Shipping Insights"
                        type="button"
                      >
                        <div
                          class="MuiBox-root css-17zsna1"
                        >
                          <p
                            class="MuiTypography-root MuiTypography-body1 css-e000bs-MuiTypography-root"
                          >
                            Shipping Insights
                          </p>
                        </div>
                      </button>
                    </div>
                  </div>
                  <div
                    class="makeStyles-menuItems-2 MuiBox-root css-1opalb8"
                  >
                    <button
                      aria-expanded="false"
                      aria-haspopup="menu"
                      class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-ipszf5-MuiButtonBase-root-MuiButton-root"
                      tabindex="0"
                      type="button"
                    >
                      Rx Workflow
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                        data-testid="KeyboardArrowDownIcon"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M7.41 8.59 12 13.17l4.59-4.58L18 10l-6 6-6-6 1.41-1.41z"
                        />
                      </svg>
                      <span
                        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                      />
                    </button>
                    <div
                      class="dropdown browserSpecific MuiBox-root css-1hyfx7x"
                    >
                      <button
                        class="subMenuList MuiBox-root css-0"
                        id="Shipped Report"
                        type="button"
                      >
                        <div
                          class="MuiBox-root css-17zsna1"
                        >
                          <p
                            class="MuiTypography-root MuiTypography-body1 css-e000bs-MuiTypography-root"
                          >
                            Shipped Report
                          </p>
                        </div>
                      </button>
                      <button
                        class="subMenuList MuiBox-root css-0"
                        id="Cancellations Report"
                        type="button"
                      >
                        <div
                          class="MuiBox-root css-17zsna1"
                        >
                          <p
                            class="MuiTypography-root MuiTypography-body1 css-e000bs-MuiTypography-root"
                          >
                            Cancellations Report
                          </p>
                        </div>
                      </button>
                      <button
                        class="subMenuList MuiBox-root css-0"
                        id="Cancellations Dashboard"
                        type="button"
                      >
                        <div
                          class="MuiBox-root css-17zsna1"
                        >
                          <p
                            class="MuiTypography-root MuiTypography-body1 css-e000bs-MuiTypography-root"
                          >
                            Cancellations Dashboard
                          </p>
                        </div>
                      </button>
                      <button
                        class="subMenuList MuiBox-root css-0"
                        id="Filling Timeliness Report"
                        type="button"
                      >
                        <div
                          class="MuiBox-root css-17zsna1"
                        >
                          <p
                            class="MuiTypography-root MuiTypography-body1 css-e000bs-MuiTypography-root"
                          >
                            Filling Timeliness Report
                          </p>
                        </div>
                      </button>
                    </div>
                  </div>
                </div>
              </div>
              <div
                class="MuiBox-root css-m18vu2"
              >
                <span
                  class="MuiBox-root css-0"
                />
                <div
                  class="MuiBox-root css-5nwj3y"
                >
                  <button
                    class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall css-20mkl5-MuiButtonBase-root-MuiIconButton-root"
                    data-testid="initials-avatar"
                    tabindex="0"
                    type="button"
                  >
                    <div
                      aria-label=""
                      class="MuiAvatar-root MuiAvatar-circular MuiAvatar-colorDefault css-jvbni9-MuiAvatar-root"
                      data-mui-internal-clone-element="true"
                    />
                    <span
                      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                    />
                  </button>
                </div>
              </div>
            </div>
          </header>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-b8oxpr-MuiGrid-root"
        >
          <h2
            class="MuiTypography-root MuiTypography-h2 css-1e9jmk9-MuiTypography-root"
          >
            An error occurred
          </h2>
          <p
            class="MuiTypography-root MuiTypography-body1 css-1gnjemi-MuiTypography-root"
          >
            A component suspended while responding to synchronous input. This will cause the UI to be replaced with a loading indicator. To fix, updates that suspend should be wrapped with startTransition.
          </p>
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium css-1cukyn2-MuiButtonBase-root-MuiButton-root"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiButton-startIcon MuiButton-iconSizeMedium css-gcc2o7-MuiButton-startIcon"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-x79i93-MuiSvgIcon-root"
                data-testid="RefreshIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M17.65 6.35C16.2 4.9 14.21 4 12 4c-4.42 0-7.99 3.58-7.99 8s3.57 8 7.99 8c3.73 0 6.84-2.55 7.73-6h-2.08c-.82 2.33-3.04 4-5.65 4-3.31 0-6-2.69-6-6s2.69-6 6-6c1.66 0 3.14.69 4.22 1.78L13 11h7V4l-2.35 2.35z"
                />
              </svg>
            </span>
            Try again
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
     
    <div
      style="height: 100vh; background: rgb(246, 245, 247);"
    >
      <div />
      <div
        class="MuiBox-root css-0"
      >
        <header
          class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionRelative css-c0mv2e-MuiPaper-root-MuiAppBar-root"
          style="border: 1px; border-color: #0000000d; border-bottom: 1px solid #0000000D; margin: 0px; box-shadow: rgb(0 0 0 / 20%) 0px 2px 4px -1px, rgb(0 0 0 / 14%) 0px 4px 5px 0px, rgb(0 0 0 / 12%) 0px 1px 10px 0px; background: rgb(255, 255, 255); z-index: 1201; font-family: \\"Roboto\\", sans-serif; width: 100%;"
        >
          <div
            class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-bw3mp6-MuiToolbar-root"
            style="padding-left: 20px; padding-right: 20px;"
          >
            <div
              class="MuiBox-root css-17zsna1"
            >
              <div
                class="MuiTypography-root MuiTypography-h6 MuiTypography-noWrap css-1m9344j-MuiTypography-root"
              >
                <a
                  aria-current="page"
                  class="active"
                  href="/"
                >
                  <img
                    alt="NexiaLogo"
                    src="test"
                    style="width: 100%; max-width: 163px; max-height: 41px;"
                  />
                </a>
              </div>
              <hr
                class="MuiDivider-root MuiDivider-middle MuiDivider-vertical MuiDivider-flexItem css-my8t4q-MuiDivider-root"
              />
              <div
                class="MuiBox-root css-bqedkg"
                style="text-decoration: none; color: rgb(101, 19, 240); font-size: 18px; margin-left: 24px;"
              >
                Enterprise Analytics
              </div>
              <div
                class="MuiBox-root css-k008qs"
              >
                <div
                  class="makeStyles-menuItems-2 MuiBox-root css-1opalb8"
                >
                  <button
                    aria-expanded="false"
                    aria-haspopup="menu"
                    class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-ipszf5-MuiButtonBase-root-MuiButton-root"
                    tabindex="0"
                    type="button"
                  >
                    Shipping Insights
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                      data-testid="KeyboardArrowDownIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M7.41 8.59 12 13.17l4.59-4.58L18 10l-6 6-6-6 1.41-1.41z"
                      />
                    </svg>
                    <span
                      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                    />
                  </button>
                  <div
                    class="dropdown browserSpecific MuiBox-root css-1hyfx7x"
                  >
                    <button
                      class="subMenuList MuiBox-root css-0"
                      id="SLA Dashboard"
                      type="button"
                    >
                      <div
                        class="MuiBox-root css-17zsna1"
                      >
                        <p
                          class="MuiTypography-root MuiTypography-body1 css-e000bs-MuiTypography-root"
                        >
                          SLA Dashboard
                        </p>
                      </div>
                    </button>
                    <button
                      class="subMenuList MuiBox-root css-0"
                      id="SLA Report"
                      type="button"
                    >
                      <div
                        class="MuiBox-root css-17zsna1"
                      >
                        <p
                          class="MuiTypography-root MuiTypography-body1 css-e000bs-MuiTypography-root"
                        >
                          SLA Report
                        </p>
                      </div>
                    </button>
                    <button
                      class="subMenuList MuiBox-root css-0"
                      id="Service Level Dashboard"
                      type="button"
                    >
                      <div
                        class="MuiBox-root css-17zsna1"
                      >
                        <p
                          class="MuiTypography-root MuiTypography-body1 css-e000bs-MuiTypography-root"
                        >
                          Service Level Dashboard
                        </p>
                      </div>
                    </button>
                    <button
                      class="subMenuList MuiBox-root css-0"
                      id="Shipping Insights"
                      type="button"
                    >
                      <div
                        class="MuiBox-root css-17zsna1"
                      >
                        <p
                          class="MuiTypography-root MuiTypography-body1 css-e000bs-MuiTypography-root"
                        >
                          Shipping Insights
                        </p>
                      </div>
                    </button>
                  </div>
                </div>
                <div
                  class="makeStyles-menuItems-2 MuiBox-root css-1opalb8"
                >
                  <button
                    aria-expanded="false"
                    aria-haspopup="menu"
                    class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-ipszf5-MuiButtonBase-root-MuiButton-root"
                    tabindex="0"
                    type="button"
                  >
                    Rx Workflow
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                      data-testid="KeyboardArrowDownIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M7.41 8.59 12 13.17l4.59-4.58L18 10l-6 6-6-6 1.41-1.41z"
                      />
                    </svg>
                    <span
                      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                    />
                  </button>
                  <div
                    class="dropdown browserSpecific MuiBox-root css-1hyfx7x"
                  >
                    <button
                      class="subMenuList MuiBox-root css-0"
                      id="Shipped Report"
                      type="button"
                    >
                      <div
                        class="MuiBox-root css-17zsna1"
                      >
                        <p
                          class="MuiTypography-root MuiTypography-body1 css-e000bs-MuiTypography-root"
                        >
                          Shipped Report
                        </p>
                      </div>
                    </button>
                    <button
                      class="subMenuList MuiBox-root css-0"
                      id="Cancellations Report"
                      type="button"
                    >
                      <div
                        class="MuiBox-root css-17zsna1"
                      >
                        <p
                          class="MuiTypography-root MuiTypography-body1 css-e000bs-MuiTypography-root"
                        >
                          Cancellations Report
                        </p>
                      </div>
                    </button>
                    <button
                      class="subMenuList MuiBox-root css-0"
                      id="Cancellations Dashboard"
                      type="button"
                    >
                      <div
                        class="MuiBox-root css-17zsna1"
                      >
                        <p
                          class="MuiTypography-root MuiTypography-body1 css-e000bs-MuiTypography-root"
                        >
                          Cancellations Dashboard
                        </p>
                      </div>
                    </button>
                    <button
                      class="subMenuList MuiBox-root css-0"
                      id="Filling Timeliness Report"
                      type="button"
                    >
                      <div
                        class="MuiBox-root css-17zsna1"
                      >
                        <p
                          class="MuiTypography-root MuiTypography-body1 css-e000bs-MuiTypography-root"
                        >
                          Filling Timeliness Report
                        </p>
                      </div>
                    </button>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="MuiBox-root css-m18vu2"
            >
              <span
                class="MuiBox-root css-0"
              />
              <div
                class="MuiBox-root css-5nwj3y"
              >
                <button
                  class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall css-20mkl5-MuiButtonBase-root-MuiIconButton-root"
                  data-testid="initials-avatar"
                  tabindex="0"
                  type="button"
                >
                  <div
                    aria-label=""
                    class="MuiAvatar-root MuiAvatar-circular MuiAvatar-colorDefault css-jvbni9-MuiAvatar-root"
                    data-mui-internal-clone-element="true"
                  />
                  <span
                    class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                  />
                </button>
              </div>
            </div>
          </div>
        </header>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-b8oxpr-MuiGrid-root"
      >
        <h2
          class="MuiTypography-root MuiTypography-h2 css-1e9jmk9-MuiTypography-root"
        >
          An error occurred
        </h2>
        <p
          class="MuiTypography-root MuiTypography-body1 css-1gnjemi-MuiTypography-root"
        >
          A component suspended while responding to synchronous input. This will cause the UI to be replaced with a loading indicator. To fix, updates that suspend should be wrapped with startTransition.
        </p>
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium css-1cukyn2-MuiButtonBase-root-MuiButton-root"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-startIcon MuiButton-iconSizeMedium css-gcc2o7-MuiButton-startIcon"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-x79i93-MuiSvgIcon-root"
              data-testid="RefreshIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M17.65 6.35C16.2 4.9 14.21 4 12 4c-4.42 0-7.99 3.58-7.99 8s3.57 8 7.99 8c3.73 0 6.84-2.55 7.73-6h-2.08c-.82 2.33-3.04 4-5.65 4-3.31 0-6-2.69-6-6s2.69-6 6-6c1.66 0 3.14.69 4.22 1.78L13 11h7V4l-2.35 2.35z"
              />
            </svg>
          </span>
          Try again
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
